parking-lot-help-request/firmware/README.md

```diff
@@ -0,0 +1,105 @@
+
+![Pasting in the Notecard config script in the in-browser Notecard playground](./readme-notecard-playground-script.png)
+
+Now, click the button a few times and hopefully you'll see the LED light on the board flash and the button press data similar to this screenshot below flowing into your own Notehub project.
+
+![Press that button and watch the LED light up](readme-press-the-button.png)
+
+https://user-images.githubusercontent.com/20400845/197624207-36afcfb9-6307-4dc5-acb9-542bec0140cf.mov
+
+![_button.qo events registering in Notehub when the help request button is pressed on the device](readme-notehub-events.png)
+_The `_button.qo` events are the events we'll want to use to trigger SMS alerts._
+
+### Transform Notehub data with JSONata for Twilio
+
+After button press events are successfully being sent to your Notehub project, they'll need to be transformed and routed out to Twilio to trigger SMS alerts to a user's phone.
+
+> **NOTE:** Although this project is using Twilio for SMS alerts, you can route your button presses to any service you'd like, and set up whatever sort of notification system you have in mind.
+
+Follow the [Twilio starter guide](https://dev.blues.io/guides-and-tutorials/twilio-sms-guide/#configuring-the-route) on the Blues Wireless developer experience site to go through the basics of setting up Twilio account, and a route from Notehub to Twilio, until you reach step 8.
+
+Instead of selecting `twilio.qo` files, you're going to want to route out `_button.qo` events in the **Filters** section.
+
+![Selecting just _button.qo events to route to Twilio from Notehub](readme-button-events.png)
+_Select just the `_button.qo` events to route out of Notehub to Twilio_
+
+Then, use [JSONata](https://dev.blues.io/guides-and-tutorials/notecard-guides/using-jsonata-to-transform-json/) (a lightweight query and transformation language for JSON) to filter down and format the data from a button event before sending it to Twilio.
+
+Twilio expects very specific syntax, and you can add the JSONata expression right at the bottom of the new route page you're working in under the **Data** section.
+
+![JSONata expression to ](readme-notehub-jsonata.png)
+
+Paste this JSONata into the Tranform Data input, and replace the `$from` and `$to` values with your Twilio number (`$from`) and your cell phone number to receive the SMS messages (`$to`).
+
+```json
+(
+    $from := "+1800XXXXXXX";
+    $to := "+1404XXXXXXX";
+    $best_device_id := best_id ? best_id : device;
+    $body := function(){
+        (
+            $join([
+                "Alert! ",
+                $best_device_id,
+                " requires assistance. Help requested at ",
+                $fromMillis(when * 1000,
+                "[M01]/[D01]/[Y0001] [h#1]:[m01][P]",
+                "-0400"),
+                " ET."
+            ])
+        )
+    };
+    $result := "&Body=" & $body() & "&From=" & $from & "&To=" & $to & "&";
+)
```

Comment:
Unneeded function+function call. I like the use of `join` though. Tested this new syntax.

```suggestion
(
    $from := "+1800XXXXXXX";
    $to := "+1404XXXXXXX";
    $best_device_id := best_id ? best_id : device;
    $body :=
            $join([
                "Alert! ",
                $best_device_id,
                " requires assistance. Help requested at ",
                $fromMillis(when * 1000,
                "[M01]/[D01]/[Y0001] [h#1]:[m01][P]",
                "-0400"),
                " EDT."
            ]);
    $result := "&Body=" & $body & "&From=" & $from & "&To=" & $to & "&";
)
```

Reply:
Good to know. JSONata's quite a useful language